src/openrct2/windows/loadsave.c

```diff
@@ -508,9 +510,20 @@ static void window_loadsave_scrollpaint(rct_window *w, rct_drawpixelinfo *dpi, s
 			continue;
 
 		rct_string_id stringId = STR_BLACK_STRING;
+		
+		// Fill colour for current list element
+		// Hovering
 		if (i == w->selected_list_item) {
 			stringId = STR_WINDOW_COLOUR_2_STRINGID;
-			gfx_filter_rect(dpi, 0, y, 800, y + 9, PALETTE_DARKEN_1);
+			gfx_fill_rect(dpi, 0, y, listWidth, y + 9, ColourMapA[w->colours[1]].darker | 0x1000000);
```

Comment:
`w->colours[1]` can be larger than `COLOUR_COUNT`, because of the transparency bit. use `ColourMapA[BASE_COLOUR(w->colours[1])].darker` to prevent this.